src/bootstrap/compile.rs

```diff
@@ -668,7 +668,7 @@ impl Step for Rustc {
         }
 
         let is_collecting = if let Some(path) = &builder.config.rust_profile_generate {
-            if compiler.stage == 1 {
+            if !builder.llvm_link_shared() || compiler.stage >= 2 {
```

Comment:
Please add a comment explaining why `llvm_link_shared` is important.

Reply:
Done